src/main/java/nl/cherement/jak/config/JWTAuthorizationFilter.java

```diff
@@ -24,37 +23,41 @@ public JWTAuthorizationFilter(AuthenticationManager authManager) {
     }
 
     @Override
-    protected void doFilterInternal(HttpServletRequest req,
-                                    HttpServletResponse res,
-                                    FilterChain chain) throws IOException, ServletException {
-        String header = req.getHeader(HEADER_STRING);
+    protected void doFilterInternal(HttpServletRequest req, HttpServletResponse res, FilterChain chain)
+            throws IOException, ServletException {
+        String token = getToken(req);
 
-        if (header == null || !header.startsWith(TOKEN_PREFIX)) {
-            chain.doFilter(req, res);
-
-            return;
+        if (token != null) {
+            SecurityContextHolder.getContext().setAuthentication(getAuthentication(token));
         }
 
-        UsernamePasswordAuthenticationToken authentication = getAuthentication(req);
-
-        SecurityContextHolder.getContext().setAuthentication(authentication);
         chain.doFilter(req, res);
     }
 
-    private UsernamePasswordAuthenticationToken getAuthentication(HttpServletRequest request) {
-        String token = request.getHeader(HEADER_STRING);
-        if (token != null) {
-            // parse the token.
+    String getToken(HttpServletRequest req) {
+        String token = req.getHeader(HEADER_STRING);
+
+        if (token == null || !token.startsWith(TOKEN_PREFIX)) {
+            String protocols = req.getHeader(WEBSOCKET_HEADER_STRING);
+            if (protocols == null || !protocols.contains(WEBSOCKET_TOKEN_PREFIX)) {
+                return null;
+            }
+            return protocols.substring(protocols.indexOf(WEBSOCKET_TOKEN_PREFIX))
+                    .replace(WEBSOCKET_TOKEN_PREFIX, "");
+        }
+        return token.replace(TOKEN_PREFIX, "");
+    }
+
+    UsernamePasswordAuthenticationToken getAuthentication(String token) {
```

Comment:
Moet deze private blijven?

Reply:
Hij kan private maar dan kunnen we hem in de test niet aanroepen. Wat een beetje vervelend is.

Reply:
Dat maakt niet zoveel uit. 
Als je de methode via een public methode aanroept en test heb je alsnog je coverage.
Push een versie van deze waarin het geval zo is.